File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        maxReach = 0
        jumps = 0
        lastJump = 0
        i = 0
        while lastJump < len(nums)-1:
            maxReach = max(maxReach, i+nums[i])
            if i == lastJump:
                lastJump = maxReach
                jumps += 1
            i+=1
        return jumps
        

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int maxReach = 0;
        int minJumps = 0;
        int lastJump = 0;
        int i = 0;
        while(lastJump < nums.length-1){
            maxReach = Math.max(maxReach, i + nums[i]);
            if(i == lastJump){
                lastJump = maxReach;
                minJumps++;
            }
            i++;
        }
        return minJumps;
    }
}